armory/data/datasets.py

```diff
@@ -237,11 +238,98 @@ def _parse(serialized_example):
     return clean_x, adv_x, labels
 
 
+def german_traffic_sign(
+    preprocessing_fn: Callable = None, dataset_dir: str = None,
+) -> (np.ndarray, np.ndarray, np.ndarray, np.ndarray):
+    """
+    German traffic sign dataset with 43 classes and over
+    50,000 images.
+
+    :param preprocessing_fn: Callable function to preprocess inputs
+    :param dataset_dir: Directory where cached datasets are stored
+    :return: (train_images,train_labels,test_images,test_labels)
+    """
+
+    from PIL import Image
+
+    def _read_images(prefix, gtFile, im_list, label_list):
+        with open(gtFile, newline="") as csvFile:
+            gtReader = csv.reader(
+                csvFile, delimiter=";"
+            )  # csv parser for annotations file
+            gtReader.__next__()  # skip header
+            # loop over all images in current annotations file
+            for row in gtReader:
+                try:
+                    tmp = Image.open(os.path.join(prefix, row[0]))
+                    # First column is filename
+                except IOError as e:
+                    raise IOError(f"Could not open image with PIL. {e}")
+                im_list.append(np.array(tmp))
+                tmp.close()
+                label_list.append(int(row[7]))  # the 8th column is the label
```

Comment:
This dataset has annotations for bounding boxes. Seeing as we don't support this (yet) I'm fine with leaving it as a classification label but down the line we'll need to modify this.